### ISSUE TITLE
Is parseText thread safe?

### ISSUE THREAD

**Issue by fabrichter** [hankcs/AhoCorasickDoubleArrayTrie#20]
Thanks for your great work on this project!
I was wondering if the implementation is thread safe when used in a read-only fashion,
i.e. I want to initalize one `AhoCorasickDoubleArrayTrie`, set it up via `build` once
and then call `parseText` from multiple threads. Is this possible / safe to do?

**Comment by hankcs** [comment]
Yes, it's safe to call `parseText` from multiple threads. As this method doesn't modify any members, but put every temporary result on stack, it's thread safe.
